artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/Packet.java

```diff
@@ -42,6 +42,21 @@ default int expectedEncodeSize() {
       return INITIAL_PACKET_SIZE;
    }
 
+   default boolean isRequiresResponse() {
```

Comment:
be careful to use `default` method on hottest paths vs common abstract methods on superclass: they lack CHA AFAIK!
http://mail.openjdk.java.net/pipermail/hotspot-compiler-dev/2015-April/thread.html#17649 and https://stackoverflow.com/questions/30312096/java-default-methods-is-slower-than-the-same-code-but-in-an-abstract-class

Reply:
good point, ill move to PacketImpl the default behaviour.

Reply:
done